include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1296,93 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * Data structure returned by
+     * GridTools::internal::distributed_compute_intersection_locations(). It can
+     * be converted to
+     * GridTools::internal::DistributedComputePointLocationsInternal, which can
+     * be used to reinit Utilities::MPI::RemotePointEvaluation.
+     */
+    template <int dim, int spacedim>
+    struct DistributedComputeIntersectionLocationsInternal
+    {
+      /**
+       * Intersection answers are assumed to be simplices (e.g. provided by
+       * CGAL)
+       */
+      using IntersectionAnswerType =
+        std::array<dealii::Point<spacedim>, dim + 1>;
+
+      /**
+       * Information of each intersection on sending/evaluation side. The
+       * elements of the tuple are as follows: 0) cell level and index, 1) rank
+       * of the owning process, 2) local index of the owning process, 3)
+       * found intersection, 4) requested intersection.
+       *
+       * @note The vector is sorted according to 1), 2).
+       */
+      std::vector<std::tuple<std::pair<int, int>,
+                             unsigned int,
+                             unsigned int,
+                             IntersectionAnswerType>>
+        send_components;
+
+      /**
+       * Ranks to send to.
+       */
+      std::vector<unsigned int> send_ranks;
```

Comment:
I would remove this...

Reply:
[a1cd90d](https://github.com/dealii/dealii/pull/15165/commits/a1cd90dfa72b54132a40dbc5b49c8d62709c98fa)